src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -9875,6 +9875,18 @@ bool Unit::Attack(Unit* victim, bool meleeAttack)
         return false;
     }
 
+    if (creature /*&& (IsHunterPet() || IsPet() || IsSummon()) && IsControlledByPlayer()*/)
+    {
+        if (victim->GetTypeId() == TYPEID_UNIT && victim->ToCreature()->IsTrigger())
```

Comment:
Isn't triggers by default not able to be attacked?

Reply:
Yes, but there is a "twitch" action